daphne_worker/src/durable/mod.rs

```diff
@@ -209,97 +190,185 @@ impl<'srv> DurableConnector<'srv> {
     }
 }
 
-trait DapDurableObject {
+/// A durable object that is capable of garbage collecting itself.
+trait GcDurableObject {
     type DurableMethod: DurableMethod;
 
-    fn state(&self) -> &State;
+    /// Instantiate the durable object from it's state and environment.
+    fn new(state: State, env: Env) -> Self;
+
+    /// Handle a durable object request.
+    async fn handle(&mut self, req: Request) -> Result<Response>;
 
-    fn deployment(&self) -> DaphneWorkerDeployment;
+    /// When this durable object should self cleanup.
+    ///
+    /// The default implementation of this function returns None, signaling that this object should
+    /// not be automatically cleaned up.
+    fn should_cleanup_at(&self) -> Option<ScheduledTime> {
+        None
```

Comment:
Consider not providing a default implementation. That way we force ourselves to think this through for each DO.